### PR TITLE
test: Add test for retrying serializable transaction

### PR DESCRIPTION
**Pull request by SevInf**
Based on #15162

To run:
1. Checkout this repo
2. `pnpm install`
3. `cd packages/client`
4. ` pnpm test:functional  9678`

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -147,7 +147,7 @@
     }
   },
   "dependencies": {
-    "@prisma/engines-version": "4.4.0-47.c6b5fdac855d9b76c03ffc33c58188b56985fc6b"
+    "@prisma/engines-version": "4.4.0-55.qe-transaction-write-conflict-911f8bae8f9b08719066939ba0aaff82c1fe4a50"
```
Comment (SevInf) [inline packages/client/package.json:150]: Note: switch to non-integration version before merge.

File: packages/client/tests/functional/issues/9678/tests.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            result = await prisma.$transaction(
+              [prisma.resource.deleteMany({ where: { name: 'name' } }), prisma.resource.createMany({ data })],
+              {
+                isolationLevel: Prisma.TransactionIsolationLevel.Serializable,
+              },
+            )
+            return result
+          } catch (e) {
+            if (e.code === 'P2034') {
+              retires++
```
Comment (janpio) [inline packages/client/tests/functional/issues/9678/tests.ts:38]:
```suggestion
        let retries = 0

        let result
        while (retries < MAX_RETRIES) {
          try {
            result = await prisma.$transaction(
              [prisma.resource.deleteMany({ where: { name: 'name' } }), prisma.resource.createMany({ data })],
              {
                isolationLevel: Prisma.TransactionIsolationLevel.Serializable,
              },
            )
            return result
          } catch (e) {
            if (e.code === 'P2034') {
              retries++
```

File: packages/client/package.json (hunk 1 above)
Comment (janpio) [reply packages/client/package.json]: Did that when solving conflicts (as I figured out that PR was merged)

**Comment by janpio** [comment]
Current state of this PR seems to depend on https://github.com/prisma/tiberius/pull/245 + https://github.com/prisma/quaint/pull/405 + https://github.com/prisma/prisma-engines/pull/3227 to be able to pass for SQL Server.

**Review by millsp — APPROVED** [review]

File: packages/client/tests/functional/issues/9678/tests.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      expect(hasRetried).toBe(true)
+    })
+  },
+  {
+    optOut: {
+      from: ['sqlite', 'mongodb', 'sqlserver'],
+      reason: `
+        sqlite - concurrent transactions are not supported
+        mongo - isolation levels are not supported
+        sqlserver - upstream work needs to be done
```
Comment (janpio) [inline packages/client/tests/functional/issues/9678/tests.ts:58]:
```suggestion
        sqlserver - upstream work needs to be done (https://github.com/prisma/tiberius/pull/245)
```

**Comment by janpio** [comment]
Follow up PR that re-enables the SQL Server tests (to show how they fail): https://github.com/prisma/prisma/pull/15558

**Pull request merged**